lms/services/canvas_api.py

```diff
@@ -114,25 +117,37 @@ def public_url(self, file_id):
         public_url_request = self._helper.public_url_request(
             self._access_token, file_id
         )
-        public_url_response = requests.Session().send(public_url_request)
+
+        try:
+            public_url_response = requests.Session().send(public_url_request)
+            public_url_response.raise_for_status()
+        except RequestException as err:
+            # TODO: Try refreshing the access token and re-trying the response.
+            response = getattr(err, "response", None)
+
+            raise CanvasAPIError(
+                explanation="Connecting to the Canvas API failed", response=response
+            ) from err
```

Comment:
Ah, cool. I didn't know Python 3 had language-level support for wrapping exceptions.